### ISSUE TITLE
Problem with parallel post fit wrapper (ParallelPostfitWrapper)--Can't drop an axis with more than 1 block. Please use atop instead

### ISSUE THREAD

**Issue by firstkingofrome** [dask/dask-ml#376]
Basically I am trying to evaluate some sklearn predictors that I came up with using standard sklearn to produce larger than memory array outputs. This code illustrates the problem:

from dask.base import tokenize
import numpy as np
import dask.array as da
from dask.array import Array
from sklearn.linear_model import LinearRegression
from dask_ml.wrappers import ParallelPostFit
"""
sample of problem
"""
x = np.linspace(0,100,100,dtype=np.int32)
y = np.linspace(0,100,100,dtype=np.int32)
z = np.linspace(0,100,100,dtype=np.int32)

Y = np.random.normal(size=(100,))
X = np.stack([x,y,z],axis=1)
reg = LinearRegression().fit(X,Y)

#now try to compute on dask arrays over the whole space
x= da.linspace(0,100,100,chunks=(10,)).astype(np.int32)
y= da.linspace(0,100,100,chunks=(10,)).astype(np.int32)
z= da.linspace(0,100,100,chunks=(10,)).astype(np.int32)
x,y,z = da.meshgrid(x,y,z,sparse=False,indexing='ij')
stacked = da.stack([x.flatten(),y.flatten(),z.flatten()],axis=1)
clf = ParallelPostFit(estimator=reg)
clf.predict(stacked)

Executing clf.predict throws a value error Can't drop an axis with more than 1 block. Please use atop instead.

which I dont understand how to correct. Thank You for any help.

**Comment by mrocklin** [comment]
I suspect that `ParallelPostFit` assumes that the data has a single block in the columns direction.  @TomAugspurger thoughts about providing an informative error here pointing users to `X.reshape({1: -1})` or doing this for them automatically?

**Comment by TomAugspurger** [comment]
That’s correct. We should at least be calling check_array, which validates this. In principle a rechunk with “auto” and -1 should do what we want.

________________________________

**Comment by firstkingofrome** [comment]
I just wanted to thank you both for how quickly you helped resolve this and for dask.

**Comment by TomAugspurger** [comment]
This will be auto-closed once #377 is merged. Just need to sort out one issue there first.
